components/ai_chat/browser/ai_chat_tab_helper.cc

```diff
@@ -0,0 +1,470 @@
+      << " previous text content but there were " << suggested_questions_.size()
+      << " suggested questions: "
+      << base::JoinString(suggested_questions_, ", ");
+
+  // Now that we have content, we can provide a summary on-demand. Add that to
+  // suggested questions.
+  // TODO(petemill): translation for this question
+  suggested_questions_.emplace_back(is_video_ ? "Summarize this video"
+                                              : "Summarize this page");
+  OnSuggestedQuestionsChanged();
+  // Automatically fetch questions related to page content, if allowed
+  bool can_auto_fetch_questions =
+      HasUserOptedIn() && is_conversation_active_ &&
+      pref_service_->GetBoolean(
+          ai_chat::prefs::kBraveChatAutoGenerateQuestions) &&
+      !article_text_.empty() && (suggested_questions_.size() <= 1);
+  if (can_auto_fetch_questions) {
+    GenerateQuestions();
+  }
+}
+
+void AIChatTabHelper::CleanUp() {
+  chat_history_.clear();
+  article_text_.clear();
+  suggested_questions_.clear();
+  is_request_in_progress_ = false;
+  has_generated_questions_ = false;
+  OnSuggestedQuestionsChanged();
+
+  // Trigger an observer update to refresh the UI.
+  for (auto& obs : observers_) {
+    obs.OnHistoryUpdate();
+  }
+}
+
+std::vector<std::string> AIChatTabHelper::GetSuggestedQuestions(
+    bool& can_generate,
+    bool& auto_generate) {
+  // Can we get suggested questions
+  can_generate = !has_generated_questions_ && !article_text_.empty();
+  // Are we allowed to auto-generate
+  auto_generate = pref_service_->GetBoolean(
+      ai_chat::prefs::kBraveChatAutoGenerateQuestions);
+  return suggested_questions_;
+}
+
+void AIChatTabHelper::GenerateQuestions() {
+  DVLOG(1) << __func__;
+  // This function should not be presented in the UI if the user has not
+  // opted-in yet.
+  DCHECK(HasUserOptedIn());
+  DCHECK(is_conversation_active_);
+  // Can't operate if we don't have an article text
+  if (article_text_.empty()) {
+    return;
+  }
+  // Don't perform the operation more than once
+  if (suggested_questions_.size() > 1u) {
+    return;
+  }
+
+  has_generated_questions_ = true;
+  OnSuggestedQuestionsChanged();
+
+  std::string prompt = base::StrCat(
+      {base::ReplaceStringPlaceholders(
+           l10n_util::GetStringUTF8(is_video_
+                                        ? IDS_AI_CHAT_VIDEO_PROMPT_SEGMENT
+                                        : IDS_AI_CHAT_ARTICLE_PROMPT_SEGMENT),
+           {article_text_}, nullptr),
+       "\n\n", l10n_util::GetStringUTF8(IDS_AI_CHAT_QUESTION_PROMPT_SEGMENT),
+       ai_chat::kAIPrompt, " <response>"});
+  // Make API request for questions.
+  // Do not call SetRequestInProgress, this progress
+  // does not need to be shown to the UI.
+  auto navigation_id_for_query = current_navigation_id_;
+  ai_chat_api_->QueryPrompt(
+      prompt,
+      base::BindOnce(
+          [](base::WeakPtr<AIChatTabHelper> tab_helper,
```

Comment:
nit: This lambda function is quite long. Perhaps we should consider abstracting it into a class function. Additionally, it might be a good idea to clean up the code when we're abstracting the data model

Reply:
moved